script.skipit/resources/lib/buildkeymap.py

```diff
@@ -0,0 +1,35 @@
+import os
+import xbmcvfs
+import xbmc
+
+
+def setup_keymap_folder():
+    if not os.path.exists(userdata):
+        os.makedirs(userdata)
+
+def write_keymap(file):
+    
+    setup_keymap_folder()
+    # if the skipit keymapfile already exists do nothing
+    if os.path.exists(file):
+        exit
+
+    keymapxml = open(file,"w")
```

Comment:
It's not allowed to edit/create files outside of your add-on's user_data folder (see https://kodi.wiki/view/Add-on_rules:

> Add-ons should store all their data in their own subfolder inside the addon_data directory. Access (read/write/delete) to any other files / folders is not allowed by default.

So unfortunately you cannot create keymaps like this.

Reply:
OK, and I understand the reason for such a rule, but it does beg the question, why have a userdata/keymaps directory if not for adding just this sort of thing.

Anyway, do you know of any other way to provide this sort of functionality, i.e. add a key binding for some addon functionality.

Currently this requires to user to use the keymap editor plugin (which does write to this directory) to map a key, which I feel is a lot less user friendly than it might be.

Reply:
They could upload a keymap.xml themselves. But it is not allowed to be done from an add-on.